lib/engine/tile.rb

```diff
@@ -288,6 +289,21 @@ def exits
       @_exits ||= @edges.map { |e| rotate(e.num, @rotation) }.uniq
     end
 
+    def converging_exit?(num)
+      exit_count[num] > 1
+    end
+
+    def exit_count
+      @_exit_count ||= compute_exit_count
```

Comment:
just make this inline

@_exit_count ||= begin
    counts = Hash.new(0) 
   @edges.each { |edge| counts[...) += 1 }
   counts